File: EbayAPI/Dtos/UserDetails.cs
using EbayAPI.Models;

namespace EbayAPI.Dtos;

public class UserDetails
{
    [Required] public string Username { get; set; }
    public string? FirstName { get; set; } = null;
    public string? LastName {get; set;} = null;
    [Required] public string Email {get; set;}
    public string? PhoneNumber {get; set;} = null;
    public string? Street {get; set;} = null;
    public int? StreetNumber {get; set;} = null;
    [Required] public string City {get; set;}
    public string? PostalCode {get; set;} = null;
    [Required] public string Country {get; set;}
    public string? VATNumber { get; set; } = null;
    public bool? Enabled { get; set; } = null;
    [Required] public int SellerRatingsNum {get; set; }
    [Required] public double SellerRating {get; set; }
    [Required] public int BidderRatingsNum {get; set; }
    [Required] public double BidderRating {get; set; }
    [Required] public DateTime DateCreated {get; set; }
}

File: EbayUI/src/app/model/UserDetails.ts
export class UserDetails {
  "username": string = "";
  "firstName"?: string = undefined;
  "lastName"?: string = undefined;
  "email": string = "";
  "phoneNumber"?: string = undefined;
  "street"?: string = undefined;
  "streetNumber"?: number = undefined;
  "city": string = "";
  "postalCode"?: string = undefined;
  "country": string = "";
  "vatNumber"?: string = undefined;
  "enabled"?: boolean = undefined;
  "sellerRatingsNum": number = 0;
  "sellerRating": number = 0;
  "bidderRatingsNum": number = 0;
  "bidderRating": number = 0;
  "dateCreated": Date;
}
